components/button/test/button.perf.js

```diff
@@ -0,0 +1,24 @@
+import '../button.js';
+import { html, testRenderTime } from 'web-test-runner-performance/browser.js';
+import { expect } from '@open-wc/testing';
+
+describe('d2l-button', () => {
+
+	it('normal', async() => {
+		const element = html`<d2l-button>Button</d2l-button>`;
+		const result = await testRenderTime(element, { iterations: 1 });
+		expect(result.duration).to.be.below(1500);
+	});
+
+	it('primary', async() => {
+		const element = html`<d2l-button primary>Button</d2l-button>`;
+		const result = await testRenderTime(element, { iterations: 1 });
+		expect(result.duration).to.be.below(1500);
+	});
+
+	it('disabled', async() => {
+		const element = html`<d2l-button disabled>Button</d2l-button>`;
+		const result = await testRenderTime(element, { iterations: 1 });
+		expect(result.duration).to.be.below(1500);
+	});
```

Comment:
I'm a little torn about where to draw the line in terms of perf testing all kinds of variations vs. sticking with the variations that we know do things "different" enough to be worth it. The more tests we have, the slower it's going to run and the higher likelihood of flake... but also the better coverage we have and the higher the chances of us catching a real performance regression.

Like for button, we know that `disabled` just puts a `disabled` attribute on the internal `<button>` element... so should have no performance implications.

Reply:
Yeah I added these for the first couple but realized they are kinda useless and didnt add them for `d2l-input-number` or `d2l-input-text`. I can just get rid of them, it'll make the test suite slightly faster anyway